scripts/tests/run_tv_casting_test.py

```diff
@@ -80,36 +87,53 @@ def dump_temporary_logs_to_console(log_file_path: str):
 
 def handle_casting_failure(casting_state: str, log_file_paths: List[str]):
     """Log '{casting_state} failed!' as error, dump log files to console, exit on error."""
-    logging.error(casting_state + ' failed!')
+    logging.error(f'{casting_state} failed!')
 
     for log_file_path in log_file_paths:
         try:
             dump_temporary_logs_to_console(log_file_path)
         except Exception as e:
-            logging.exception(f"Failed to dump {log_file_path}: {e}")
+            logging.exception(f'Failed to dump {log_file_path}: {e}')
 
     sys.exit(1)
 
 
-def extract_value_from_string(line: str) -> str:
+def extract_value_from_string(line: str, value_name: str, casting_state: str, log_paths) -> str:
     """Extract and return value from given input string.
 
     The string is expected to be in the following format as it is received
     from the Linux tv-casting-app output:
+    \x1b[0;34m[1715206773402] [20056:2842184] [DMG]  Cluster = 0x506,\x1b[0m
+    The substring to be extracted here is '0x506'.
+    Or:
+    \x1b[0;32m[1714582264602] [77989:2286038] [SVR] Discovered Commissioner #0\x1b[0m
+    The integer value to be extracted here is '0'.
+    Or:
     \x1b[0;34m[1713741926895] [7276:9521344] [DIS] Vendor ID: 65521\x1b[0m
-    The integer value to be extracted here is 65521.
+    The integer value to be extracted here is '65521'.
     Or:
     \x1b[0;34m[1714583616179] [7029:2386956] [SVR] 	device Name: Test TV casting app\x1b[0m
     The substring to be extracted here is 'Test TV casting app'.
     """
-    value = line.split(':')[-1].strip().replace('\x1b[0m', '')
+    if ':' in line:
```

Comment:
This seems odd a bit to me: the example we have a above is `\x1b[0;34m[1715206773402] [20056:2842184] [DMG] Cluster = 0x506,\x1b[0m` ... which does match the `:` but it looks like by chance since the log line staring with `[DMG]` does not actually contain the ':'. Same comment on the `#`.

It looks to me like everything seems to match.
Maybe we should have a different parsing logic. We seem to maybe search for `if value_name in line` as being a more accurate search.

Reply:
I wonder if we should have some form of regex given the format that I see: 
  - some prefix (like time, color changing)
  - ` \[[A-Z]{1,3}\]` which represents the logging source. The 1-3 characters is guaranteed by https://github.com/project-chip/connectedhomeip/blob/master/src/lib/support/logging/TextOnlyLogging.cpp#L111
   - then the log with an optional suffix of `\x1b[0m`
   Seems maybe valuable to extract the log text from a formatted log line, and then processing the log text by extracting `#,=,:`

Reply:
Good idea! Addressed in the latest commit.